src/ui/component/stepper/stepper.tsx

```diff
@@ -29,10 +35,15 @@ export const Stepper: FC<StepperProps> = ({ steps: stepElements }) => {
   const activeStepElement = stepElements.find(step => step.id === activeStepId) ?? stepElements[0]
   const activeStep = steps.find(step => step.id === activeStepId) ?? steps[0]
 
+  const isEveryStepValid = stepElements.every(stepElement => stepElement.validate?.() ?? true)
+  const isLastStep = activeStep.order === steps.length - 1
```

Comment:
I think this constants should be calculated and returned by the hook to reduce code duplication when using the stepper in several places, what do you think?

Reply:
The result would be the same because a new hook function is created for each Stepper instance.

Reply:
For me it's the same logiq as getting back `activeStepId` or `previousActiveStepId` from the hook, your new 
 constants could be calculated in the hook and be returned by it. @fredericvilcot any advice on it?

Reply:
Also, we currently have no references to `stepElement` within the hook, so we cannot call the validate functions from the hook.

Reply:
AFAIS, the hook should be improved to support step validation. I can also see a few other areas for improvement, but we'll see at a later date.

Reply:
ok so let's go :)